chainkit_client.go

```diff
@@ -0,0 +1,114 @@
+package lndclient
+
+import (
+	"bytes"
+	"context"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/lnrpc/chainrpc"
+	"google.golang.org/grpc"
+)
+
+// ChainKitClient exposes chain functionality.
+type ChainKitClient interface {
+	// GetBlock returns a block given the corresponding block hash.
+	GetBlock(ctx context.Context, hash chainhash.Hash) (*wire.MsgBlock,
+		error)
+
+	// GetBestBlock returns the latest block hash and current height of the
+	// valid most-work chain.
+	GetBestBlock(ctx context.Context) (chainhash.Hash, int32, error)
+
+	// GetBlockHash returns the hash of the block in the best blockchain
+	// at the given height.
+	GetBlockHash(ctx context.Context, blockHeight int64) (chainhash.Hash,
+		error)
+}
+
+type chainKitClient struct {
+	client   chainrpc.ChainKitClient
+	chainMac serializedMacaroon
+	timeout  time.Duration
+
+	wg sync.WaitGroup
+}
+
+func newChainKitClient(conn grpc.ClientConnInterface,
+	chainMac serializedMacaroon, timeout time.Duration) *chainKitClient {
+
+	return &chainKitClient{
+		client:   chainrpc.NewChainKitClient(conn),
+		chainMac: chainMac,
+		timeout:  timeout,
+	}
+}
+
+func (s *chainKitClient) WaitForFinished() {
+	s.wg.Wait()
+}
+
+// GetBlock returns a block given the corresponding block hash.
+func (s *chainKitClient) GetBlock(ctx context.Context,
+	hash chainhash.Hash) (*wire.MsgBlock, error) {
+
+	macaroonAuth := s.chainMac.WithMacaroonAuth(ctx)
```

Comment:
Given we have a timeout struct member, should we add a timeout to these calls? Since they might take quite a long time on Neutrino we might not want them to block indefinitely.